app/src/main/java/com/github/sdp_begreen/begreen/fragments/CameraWithUIFragment.kt

```diff
@@ -0,0 +1,331 @@
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import android.widget.ArrayAdapter
+import android.widget.AutoCompleteTextView
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.Toast
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageCapture
+import androidx.camera.core.ImageCaptureException
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.view.PreviewView
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.core.content.ContextCompat.getSystemService
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.firebase.Auth
+import com.github.sdp_begreen.begreen.firebase.DB
+import com.github.sdp_begreen.begreen.models.ParcelableDate
+import com.github.sdp_begreen.begreen.models.PhotoMetadata
+import com.github.sdp_begreen.begreen.viewModels.ConnectedUserViewModel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.koin.android.ext.android.inject
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+
+class CameraWithUIFragment : Fragment() {
+    // Get the db instance
+    private val db by inject<DB>()
+    private val auth by inject<Auth>()
+    private val connectedUserViewModel: ConnectedUserViewModel by viewModels(ownerProducer = { requireActivity() })
+    private lateinit var outputDirectory: File
+    private var viewFinder = view?.findViewById<PreviewView>(R.id.viewFinder)
+    private var imageCapture: ImageCapture? = null
+    private lateinit var cameraExecutor: ExecutorService
+    private var lensFacing = CameraSelector.DEFAULT_FRONT_CAMERA
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_camera_with_ui, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewFinder = view.findViewById(R.id.viewFinder)
+        initView()
+    }
+
+
+    /**
+     * Initialize the view
+     */
+    private fun initView() {
+        // Request camera permissions
+        if (allPermissionsGranted()) {
+            startCamera()
+        } else {
+            ActivityCompat.requestPermissions(
+                requireActivity(), REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS
+            )
```

Comment:
In fact I inspired myself from this same documentation with this code snippet:
![image](https://user-images.githubusercontent.com/56815252/235884003-35820ceb-02a3-45d4-88bf-0a18b8b98ace.png)
And I don't see it as deprecated. So, is the switch to this new version really bringing something new? If yes I'll implement it with pleasure

Reply:
Sorry, maybe I wasn't really clear in my explanation. 

This part is totally correct to check the permission that's not the problem. 

But if you look in the else part, the `requestPermissionLauncher`, and you go up just the previous box of example code, you will see that it is defined using `registerForActivityResult` not `ActivityCompat.requestPermission`. That's what I was telling you about.

If you still have question about it don't hesitate we can discuss it together it will be easier.

Reply:
Oh yeah I agree I thought you were spoken about the requestPermission line itself